docs/source/stability_policy.rst

```diff
@@ -0,0 +1,132 @@
+This covers both the :mod:`telegram` package and the :mod:`telegram.ext` package.
+
+What doesn't this policy cover?
+-------------------------------
+
+Introduction of new features or changes of flavors of comparable behavior (e.g. the default for the used HTTP protocol version) are not covered by this policy.
+The internal structure of classes in PTB, i.e. things like the result of ``dir(obj))`` or the contents of ``obj.__dict__``, may change.
+
+Objects are in general not guaranteed to be pickleable (unless stated otherwise) and pickled objects from one version of PTB may not be loadable in future versions.
+We may provide a way to convert pickled objects from one version to another, but this is not guaranteed.
+
+Functionality that is part of PTBs API but is explicitly documented to not be intended to be used directly by users (e.g. :meth:`telegram.request.BaseRequest.do_request`) may change.
+This also applies to functions or attributes marked as final in the sense of `PEP 591 <https://www.python.org/dev/peps/pep-0591/>`__.
+
+PTB has dependencies to 3rd party packages.
+The versions that PTB uses of these 3rd party packages may change if that does not change PTBs public API.
+
+PTB does not give guarantees about which Python versions are supported.
+In general, we will try to support all Python versions that have not yet reached their end of life, but we reserve ourselves the option to drop support for Python versions earlier if that benefits the advancement of the library.
+
+.. _bot-api-functionality-1:
+
+Bot API Functionality
+~~~~~~~~~~~~~~~~~~~~~
+
+Comparison of equality of the classe in the :mod:`telegram` package is subject to change and the PTB team will update the behavior to best reflect updates in the Bot API.
+
+When arguments of the Bot API methods change order or become optional/mandatory due to changes in the Bot API, PTB will always try to reflect these changes.
+While we try to make such changes backwards compatible, this is not always possible or only with significant effort.
+In such cases we will find a trade-off between backwards compatibility and fully complying with the Bot API, which may result in breaking changes.
+We highly recommend using keyword arguments, which can help to make such changes non-breaking on your end.
+
+When the Bot API changes attributes of classes, the method :meth:`telegram.TelegramObject.to_dict` will change as necessary to reflect these changes.
+In particular, attributes deprecated by Telegram will be removed from the returned dictionary.
+Deprecated attributes that are still passed by Telegram will be available in the :attr:`~telegram.TelegramObject.api_kwargs` dictionary as long as PTB can support that with feasible effort.
+Since attributes of the classes in the :mod:`telegram` package are not writable, we may change them to properties where appropriate.
+
+Development Versions
+~~~~~~~~~~~~~~~~~~~~
+
+Before a feature is in a release, it is not covered by this policy and may change.
```

Comment:
```suggestion
Before a feature is in a release, it is not covered by this policy and may change drastically between versions.
```

Reply:
no, this is about code on master branch that's not been released yet

Reply:
or about stuff in not stable releases no?

Reply:
No. non-stable releases are still releases. What I mean here is that code that hasn't made it to pypi can change at any time.

Reply:
Wait but this doesn't make sense with the next sentence then. Since pre-releases are also not covered, can't you merge it in one sentence?


_Before a feature is in a stable release, i.e. the feature was merged into the master `branch` but not released yet or it is only present in pre-releases marked as alpha, beta or release candidate, it is not covered by this policy and may change at any time._